### PR TITLE
Adding user information to the access token

### PR DESCRIPTION
**Pull request by LanternNassi**
# Description

This is to add extra user information such as the name , id , email , etc to the JWT token . This will make it possible for the other microservices to access information about a specific user

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Trello Ticket ID

https://trello.com/c/mpsKiFuw/1859-add-the-token-verification-for-the-db-service

## How Can This Be Tested?

This can be tested through decoding the access token and accessing all the information.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by inno-asiimwe — COMMENTED** [review]
This makes sense to me.

**Review by Mubangizi — COMMENTED** [review]
I don't think this is necessary @LanternNassi

**Comment by rhodinemma** [comment]
> I don't think this is necessary @LanternNassi

was in the middle of reviewing this, maybe we need to understand a better approach for identity management in our microservices.

File: server.py
Comment (KiraboIsaacBuyondo) [inline server.py:1]: This looks okay just that we can later consider selecting only the attributes we want to include in the token. we will have more control over the information we need.

@jwt.user_claims_loader
def add_claims_to_access_token(user):
    return {
        'id': user.id,
        'email': user.email,
        'name': user.name,
        # Other attributes as needed
    }

**Comment by Mubangizi** [comment]
> > I don't think this is necessary @LanternNassi
> 
> was in the middle of reviewing this, maybe we need to understand a better approach for identity management in our microservices.

Yeah, we are going to do it, I am actually not sure why this implementation is coming in before then

**Comment by LanternNassi** [comment]
Well . The reason as to why i implemented it like this is because of a certain situation we had where the db microservice couldnt access the information of the user accessing it in real time since the token issued only had the `id and roles` . So to match the verdict , a conclusion was made that i add extra information to the token to be able to work on this task https://trello.com/c/mpsKiFuw/1859-add-the-token-verification-for-the-db-service which required that info .

**Pull request closed without merging**